File: TP3/Models/LivreDAO.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.livres.dao;

import com.livres.model.Livre;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.LinkedList;
import java.util.List;

/**
 *
 * @author usager
 */
public class LivreDAO extends DAO<Livre>{

    @Override
    public boolean create(Livre x) {
        String req = "INSERT INTO book (`ISBN` , `AUTHOR` , `TITLE`, `NB_PAGES`, `EDITION`, `YEAR`, `LANGUAGE`, `DESCRIPTION`, `KEYWORDS`) VALUES (?,?,?,?,?,?,?,?,?)";
                PreparedStatement stm = null;
		try {
                    stm = connexion.getInstance().prepareStatement(req);
                    stm.setString(1,x.getISBN());
                    stm.setString(2,x.getAuthor() );
                    stm.setString(3,x.getTitle());
                    stm.setInt(4, x.getNB_Pages());
                    stm.setString(5,x.getEdition());
                    stm.setInt(6,x.getYear());
                    stm.setString(7,x.getLanguage());
                    stm.setString(8,x.getDescription());
                    stm.setString(9,x.getKeywords());
                    int n = stm.executeUpdate();
                    System.out.println("stm.executeUpdate() result: "+n);
                    if (n>0) {
                        stm.close();
                        return true;
                    }
		}
		catch (SQLException exp) {
                    exp.printStackTrace();
		}
		finally {
                    if (stm!=null)
                    try {
                        stm.close();
                    }
                    catch (SQLException e) {
                        // TODO Auto-generated catch block
                        e.printStackTrace();
                    }
		}
		return false;
    }

    @Override
    public Livre read(int id) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    @Override
    public Livre read(String id) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    @Override
    public boolean update(Livre x) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    @Override
    public boolean delete(Livre x) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    @Override
    public List<Livre> findAll() {
        List<Livre> liste = new LinkedList<>();

        try {
            Statement stm = connexion.getInstance().createStatement();
            ResultSet r = stm.executeQuery("SELECT * FROM book");
            while(r.next()){
                Livre livre = new Livre();
                livre.setISBN(r.getString("ISBN"));
                livre.setAuthor(r.getString("AUTHOR"));
                livre.setTitle(r.getString("TITLE"));
                livre.setNB_Pages(Integer.parseInt(r.getString("NB_PAGES")));
                livre.setEdition(r.getString("EDITION"));
                livre.setYear(Integer.parseInt(r.getString("YEAR")));
                livre.setLanguage(r.getString("LANGUAGE"));
                livre.setDescription(r.getString("DESCRIPTION"));
                livre.setKeywords(r.getString("KEYWORDS"));
                liste.add(livre);
            }
            r.close();
            stm.close();
        }
        catch(SQLException exp) {
            exp.printStackTrace();
        }
        return liste;
    }


}

File: TP3/Models/LivreDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MySql.Data.MySqlClient;
using System.Collections.ObjectModel;

namespace TP3.Models
{
    public class LivreDAO
    {
        private List<Livre> Livres = new List<Livre>();
        private MySqlConnection conn;

        public LivreDAO(MySqlConnection p_conn)
        {
            conn = p_conn;
        }

        public List<Livre> FindAllLivre()
        {
            MySqlDataReader rdr = null;

            string stm = "SELECT * FROM book";

            try
            {
                MySqlCommand cmd = new MySqlCommand(stm, conn);
                rdr = cmd.ExecuteReader();

                while (rdr.Read())
                {
                    Livres.Add(new Livre(rdr.GetString(0), rdr.GetString(1), rdr.GetString(2), rdr.GetInt32(3), rdr.GetString(4), rdr.GetInt32(5), rdr.GetString(6), rdr.GetString(7), rdr.GetString(8)));

                }                

            }
            catch (MySqlException ex)
            {
                System.Diagnostics.Debug.WriteLine("Error: {0}", ex.ToString());

            }

            return Livres;
        }

        public bool CreateLivre(Livre l)
        {
            try
            {
                MySqlCommand cmd = new MySqlCommand();
                cmd.Connection = conn;
                cmd.CommandText = "INSERT INTO book (ISBN , AUTHOR , TITLE, NB_PAGES, EDITION, YEAR, LANGUAGE, DESCRIPTION, KEYWORDS) VALUES (@ISBN,@AUTHOR,@TITLE,@NB_PAGES,@EDITION,@YEAR,@LANGUAGE,@DESCRIPTION,@KEYWORDS)";
                cmd.Prepare();

                cmd.Parameters.AddWithValue("@ISBN", l.isbn);
                cmd.Parameters.AddWithValue("@AUTHOR", l.author);
                cmd.Parameters.AddWithValue("@TITLE", l.title);
                cmd.Parameters.AddWithValue("@NB_PAGES", l.nbPages);
                cmd.Parameters.AddWithValue("@EDITION", l.edition);
                cmd.Parameters.AddWithValue("@YEAR", l.year);
                cmd.Parameters.AddWithValue("@LANGUAGE", l.language);
                cmd.Parameters.AddWithValue("@DESCRIPTION", l.description);
                cmd.Parameters.AddWithValue("@KEYWORDS", l.keywords);

                return cmd.ExecuteNonQuery() > 0;

            }
            catch (MySqlException ex)
            {
                System.Diagnostics.Debug.WriteLine("Error: {0}", ex.ToString());
            }
            return false;

        }
    }
}